documentation/proposals/Proposal - Silk.NET 3.X Generic Maths.md

```diff
@@ -0,0 +1,884 @@
+    public static implicit operator Vector2D<T>((T X, T Y) components);
+    #endregion
+
+    public void Deconstruct(out T x, out T y);
+
+    /// <summary>Returns a vector whose elements are the absolute values of each of the specified vector's elements.</summary>
+    /// <param name="value">A vector.</param>
+    /// <returns>The absolute value vector.</returns>
+    static Vector2D<T> INumberBase<Vector2D<T>>.Abs(Vector2D<T> value);
+    static Vector2D<T> IParsable<Vector2D<T>>.Parse(string s, IFormatProvider? provider);
+    static Vector2D<T> ISpanParsable<Vector2D<T>>.Parse(ReadOnlySpan<char> s, IFormatProvider? provider);
+    public static Vector2D<T> Parse(string s, NumberStyles style = default, IFormatProvider? provider = null);
+    public static Vector2D<T> Parse(ReadOnlySpan<char> s, NumberStyles style = NumberStyles.None, IFormatProvider? provider = null);
+    public static bool TryParse(string? s, IFormatProvider? provider, out Vector2D<T> result);
+    public static bool TryParse(string? s, NumberStyles style, IFormatProvider? provider, out Vector2D<T> result);
+    public static bool TryParse(ReadOnlySpan<char> s, IFormatProvider? provider, out Vector2D<T> result);
+    public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out Vector2D<T> result);
+
+    static bool INumberBase<Vector2D<T>>.IsCanonical(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsComplexNumber(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsEvenInteger(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsFinite(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsImaginaryNumber(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsInfinity(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsInteger(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsNaN(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsNegative(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsNegativeInfinity(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsNormal(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsOddInteger(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsPositive(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsPositiveInfinity(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsRealNumber(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsSubnormal(Vector2D<T> value);
+    static bool INumberBase<Vector2D<T>>.IsZero(Vector2D<T> value);
+    static Vector2D<T> INumberBase<Vector2D<T>>.MaxMagnitude(Vector2D<T> x, Vector2D<T> y);
+    static Vector2D<T> INumberBase<Vector2D<T>>.MaxMagnitudeNumber(Vector2D<T> x, Vector2D<T> y);
+    static Vector2D<T> INumberBase<Vector2D<T>>.MinMagnitude(Vector2D<T> x, Vector2D<T> y);
+    static Vector2D<T> INumberBase<Vector2D<T>>.MinMagnitudeNumber(Vector2D<T> x, Vector2D<T> y);
```

Comment:
Why no Min/Max? These are notably APIs we're going to be exposing even on the non generic System.Numerics.Vector2/3/4 types

Reply:
Min/Max are elsewhere in the spec. These are just to implement INumberBase.

Reply:
🤔 I thought I saw these when I reviewed, unless this has changed since.